src/utils/retry.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {HttpErrors} from '@loopback/rest';
+import {promisify} from 'util';
+import * as debugFactory from 'debug';
+const debug = debugFactory('loopback:example:shopping');
+
+export interface TaskStatus<T> {
+  done: boolean;
+  value?: T | null;
```

Comment:
Nitpick: this type allows the developer to provide combinations of done/value that are not valid:

 - `{done: true}` (Missing value. What should be the outcome of the operation?)
 - `{done: false, value: 'some data'}`